app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,373 @@
+                    let (_, hover_color,selected_color,greyed_out_color) = colors;
+                    Params {
+                        margin,
+                        text_offset: *text_offset,
+                        text_size: text::Size::from(*text_size),
+                        hover_color:*hover_color,
+                        font: ImString::new(font),
+                        selected_color: *selected_color,
+                        highlight_corners_radius,
+                        greyed_out_color: *greyed_out_color,
+                        greyed_out_start
+                    }
+                }
+            );
+            params_sampler <- params.sampler();
+        }
+        params_sampler
+    }
+}
+
+// === Entry ===
+
+/// A Breadcrumbs entry.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Entry {
+    frp:  EntryFrp<Self>,
+    data: Rc<EntryData>,
+}
+
+impl ensogl_grid_view::Entry for Entry {
+    type Model = Model;
+    type Params = Params;
+
+    fn new(app: &Application, text_layer: Option<&Layer>) -> Self {
+        let data = Rc::new(EntryData::new(app, text_layer));
+        let frp = EntryFrp::<Self>::new();
+        let input = &frp.private().input;
+        let out = &frp.private().output;
+        let network = frp.network();
+        let color_anim = Animation::new(network);
+        let appear_anim = Animation::new(network);
+        fn mix(c1: &color::Rgba, c2: &color::Rgba, coefficient: &f32) -> color::Rgba {
+            color::mix(*c1, *c2, *coefficient)
+        }
+
+        enso_frp::extend! { network
+            init <- source_();
+            size <- input.set_size.on_change();
+            margin <- input.set_params.map(|p| p.margin).on_change();
+            hover_color <- input.set_params.map(|p| p.hover_color).on_change();
+            font <- input.set_params.map(|p| p.font.clone_ref()).on_change();
+            text_offset <- input.set_params.map(|p| p.text_offset).on_change();
+            text_color <- input.set_params.map(|p| p.selected_color).on_change();
+            text_size <- input.set_params.map(|p| p.text_size).on_change();
+            greyed_out_color <- input.set_params.map(|p| p.greyed_out_color).on_change();
+            greyed_out_from <- input.set_params.map(|p| p.greyed_out_start).on_change();
+            highlight_corners_radius <- input.set_params.map(|p| p.highlight_corners_radius).on_change();
+            transparent_color <- init.constant(color::Rgba::transparent());
+
+            col <- input.set_location._1();
+            should_grey_out <- all_with(&col, &greyed_out_from,
+                |col, from| from.map_or(false, |from| *col >= from)
+            );
+            color_anim.target <+ should_grey_out.map(|should| if *should { 1.0 } else { 0.0 });
+            target_color <- all_with3(&text_color, &greyed_out_color, &color_anim.value, mix);
+            appear_anim.target <+ init.constant(1.0);
```

Comment:
very nice FRP names here - as I mentioned before, I like the FRP look in this PR!